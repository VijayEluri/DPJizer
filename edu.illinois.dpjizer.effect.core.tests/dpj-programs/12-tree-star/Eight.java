/**
 * This file is licensed under the University of Illinois/NCSA Open Source License. See LICENSE.TXT for details.
 */
/**
 * C2.m2 -> C1.m1 where {Y/X}
 * C2.m1 -> C2.m2 where {X/Y:L} 
 */
package eight;

/**
 * 
 * @author Mohsen Vakilian
 * 
 */
class C1<region X> {
	int a in X;

	C1() {}

	void m1() {
		a = 1;
		new C2<X>().m2();
	}

}

class C2<region Y> {
	region L;
	int b in Y;

	C2() {}

	void m2() {
		b = 2;
		new C1<Y:L>().m1();
	}
}