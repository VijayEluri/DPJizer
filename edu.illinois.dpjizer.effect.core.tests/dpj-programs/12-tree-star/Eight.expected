eight.C1.C1() invokes java.lang.Object.Object()
eight.C1.m1() writes X invokes eight.C2.m2() where { this -> X : *, Y -> X }
eight.C2.C2() invokes java.lang.Object.Object()
eight.C2.m2() writes Y invokes eight.C1.m1() where { this -> Y : eight.C2.L : *, X -> Y : eight.C2.L }
